### ISSUE TITLE
Question: Why do PluginError have a `cause` argument to the constructor ?

### ISSUE THREAD

**Issue by Carreau** [napari/napari-plugin-engine#47]
It seem to  like this is trying to do the same as 

```
except ... as e:
     raise PluginError(...) from e
```

The "from" seem to set the `__cause__` the same way the argument does.

It looks like @tlambert03 wrote the code, maybe he knows ?

**Comment by tlambert03** [comment]
yes, it's doing the same thing.  Recall though that one of the main designs of plugin errors was to _not_ raise them, so adding `__cause__` is the same as `raising from`, but without raising.   If you'd prefer to raise/catch rather than do it like this, as you did over in the napari PRs, that's fine... same ultimate goal

**Comment by Carreau** [comment]
Main problem is that if not raising PluginErrors,  then the ExecInfo will not have a `__traceback__` which will crash the IPython traceback formatter in IPython 8.0+. So I'm fine with the `PluginError(cause=e)`, but I'm a bit worried about the pattern of creating plugin errors that are exception w/o raising them

**Comment by tlambert03** [comment]
Yeah, I saw that.  Like I said, I suspect you could also add a `__traceback__` just as I added `__cause__` ... but it really doesn't matter to me.  whatever you want to do works for me.
